### ISSUE TITLE
5.0.0 and 5.0.1 reports false positives on IDEA 2016.3.3

### ISSUE THREAD

**Issue by cjr8020** [jshiell/checkstyle-idea#287]
CheckStyle-IDEA plugin versions 5.0.0 and 5.0.1 seem broken on IDEA 2016.3.3 community edition

"CheckProjects" lists classes which are false positives.   
Navigation also seems to be broken - clicking on any class listed as having problems does not open that class in the editor.   This is a Java/Maven project.
Reverting back to 4.35.0 - no problems.

**Comment by jshiell** [comment]
Thanks for the report and apologies for the breakage - I'll try get it sorted later this week.

**Comment by jrumbinas** [comment]
The following property is ignored in 5.0.1 (4.35.0 - works as expected)
`<property name="severity" value="ignore"/>`

**Comment by jshiell** [comment]
@jrumbinas thanks for the reports. That sounds like a different issue though - can you please open a new issue; tell me which version of Checkstyle you're running against; and attach your rules file? Thanks.

@cjr8020 could you please attach your rules file to this issue and let me know what version of Checkstyle you're running against? Thanks.

**Comment by cjr8020** [comment]
Here is the contents of my checkstyle rules file. 
Same behavior with Checkstyle-IDEA 5.0.1, reverting back to 4.35.0 for now.

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<!--

    Checkstyle configuration that checks the Google coding conventions from:

    -  Google Java Style
       https://google-styleguide.googlecode.com/svn-history/r130/trunk/javaguide.html

    Checkstyle is very configurable. Be sure to read the documentation at
    http://checkstyle.sf.net (or in your downloaded distribution).

    Most Checks are configurable, be sure to consult the documentation.

    To completely disable a check, just comment it out or delete it from the file.

    Authors: Max Vetrenko, Ruslan Diachenko, Roman Ivanov.

 -->
<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <property name="severity" value="warning"/>

    <property name="fileExtensions" value="java, properties, xml"/>
    <!-- Checks for whitespace                               -->
    <!-- See http://checkstyle.sf.net/config_whitespace.html -->
        <module name="FileTabCharacter">
            <property name="eachLine" value="true"/>
        </module>
    <!--
        enables filter to suppress audit events
        between a comment containing CHECKSTYLE:OFF
        and a comment containing CHECKSTYLE:ON:
    -->
    <module name="SuppressionCommentFilter"/>
    <!--
        enables a filter to suppress audit events for check
        on any line with a comment CHECKSTYLE IGNORE THIS LINE
    -->
    <module name="SuppressWithNearbyCommentFilter">
        <property name="commentFormat" value="CHECKSTYLE IGNORE THIS LINE"/>
        <property name="checkFormat" value=".*"/>
        <property name="influenceFormat" value="1"/>
    </module>

    <module name="TreeWalker">
        <module name="OuterTypeFilename"/>
        <module name="IllegalTokenText">
            <property name="tokens" value="STRING_LITERAL, CHAR_LITERAL"/>
            <property name="format" value="\\u00(08|09|0(a|A)|0(c|C)|0(d|D)|22|27|5(C|c))|\\(0(10|11|12|14|15|42|47)|134)"/>
            <property name="message" value="Avoid using corresponding octal or Unicode escape."/>
        </module>
        <module name="AvoidEscapedUnicodeCharacters">
            <property name="allowEscapesForControlCharacters" value="true"/>
            <property name="allowByTailComment" value="true"/>
            <property name="allowNonPrintableEscapes" value="true"/>
        </module>
        <module name="LineLength">
            <property name="max" value="160"/>
            <property name="ignorePattern" value="^package.*|^import.*|a href|href|http://|https://|ftp://"/>
        </module>
        <module name="AvoidStarImport"/>
        <module name="OneTopLevelClass"/>
        <module name="NoLineWrap"/>
        <module name="EmptyBlock">
            <property name="option" value="TEXT"/>
            <property name="tokens" value="LITERAL_TRY, LITERAL_FINALLY, LITERAL_IF, LITERAL_ELSE, LITERAL_SWITCH"/>
        </module>
        <module name="NeedBraces"/>
        <module name="LeftCurly">
            <property name="maxLineLength" value="100"/>
        </module>
        <module name="RightCurly"/>
        <module name="RightCurly">
            <property name="option" value="alone"/>
            <property name="tokens" value="CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, LITERAL_WHILE, LITERAL_DO, STATIC_INIT, INSTANCE_INIT"/>
        </module>
        <module name="WhitespaceAround">
            <property name="allowEmptyConstructors" value="true"/>
            <property name="allowEmptyMethods" value="true"/>
            <property name="allowEmptyTypes" value="true"/>
            <property name="allowEmptyLoops" value="true"/>
            <message key="ws.notFollowed" value="WhitespaceAround: ''{0}'' is not followed by whitespace. Empty blocks may only be represented as '{}' when not part of a multi-block statement (4.1.3)"/>
             <message key="ws.notPreceded" value="WhitespaceAround: ''{0}'' is not preceded with whitespace."/>
        </module>
        <module name="OneStatementPerLine"/>
        <module name="MultipleVariableDeclarations"/>
        <module name="ArrayTypeStyle"/>
        <module name="MissingSwitchDefault"/>
        <module name="FallThrough"/>
        <module name="UpperEll"/>
        <module name="ModifierOrder"/>
        <module name="EmptyLineSeparator">
            <property name="allowNoEmptyLineBetweenFields" value="true"/>
        </module>
        <module name="SeparatorWrap">
            <property name="tokens" value="DOT"/>
            <property name="option" value="nl"/>
        </module>
        <module name="SeparatorWrap">
            <property name="tokens" value="COMMA"/>
            <property name="option" value="EOL"/>
        </module>
        <module name="PackageName">
            <property name="format" value="^[a-z]+(\.[a-z][a-z0-9]*)*$"/>
            <message key="name.invalidPattern" value="Package name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="TypeName">
            <message key="name.invalidPattern" value="Type name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="MemberName">
            <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9]*$"/>
            <message key="name.invalidPattern" value="Member name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="ParameterName">
            <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9]*$"/>
            <message key="name.invalidPattern" value="Parameter name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="LocalVariableName">
            <property name="tokens" value="VARIABLE_DEF"/>
            <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9]*$"/>
            <property name="allowOneCharVarInForLoop" value="true"/>
            <message key="name.invalidPattern" value="Local variable name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="ClassTypeParameterName">
            <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
            <message key="name.invalidPattern" value="Class type name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="MethodTypeParameterName">
            <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
            <message key="name.invalidPattern" value="Method type name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="NoFinalizer"/>
        <module name="GenericWhitespace">
            <message key="ws.followed" value="GenericWhitespace ''{0}'' is followed by whitespace."/>
             <message key="ws.preceded" value="GenericWhitespace ''{0}'' is preceded with whitespace."/>
             <message key="ws.illegalFollow" value="GenericWhitespace ''{0}'' should followed by whitespace."/>
             <message key="ws.notPreceded" value="GenericWhitespace ''{0}'' is not preceded with whitespace."/>
        </module>
        <module name="Indentation">
            <property name="basicOffset" value="2"/>
            <property name="braceAdjustment" value="0"/>
            <property name="caseIndent" value="2"/>
            <property name="throwsIndent" value="4"/>
            <property name="lineWrappingIndentation" value="4"/>
            <property name="arrayInitIndent" value="2"/>
        </module>
        <module name="AbbreviationAsWordInName">
            <property name="ignoreFinal" value="false"/>
            <property name="allowedAbbreviationLength" value="1"/>
        </module>
        <module name="OverloadMethodsDeclarationOrder"/>
        <module name="VariableDeclarationUsageDistance"/>
        <module name="MethodParamPad"/>
        <module name="OperatorWrap">
            <property name="option" value="NL"/>
            <property name="tokens" value="BAND, BOR, BSR, BXOR, DIV, EQUAL, GE, GT, LAND, LE, LITERAL_INSTANCEOF, LOR, LT, MINUS, MOD, NOT_EQUAL, PLUS, QUESTION, SL, SR, STAR "/>
        </module>
        <module name="AnnotationLocation">
            <property name="tokens" value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF, METHOD_DEF, CTOR_DEF"/>
        </module>
        <module name="AnnotationLocation">
            <property name="tokens" value="VARIABLE_DEF"/>
            <property name="allowSamelineMultipleAnnotations" value="true"/>
        </module>
        <module name="MethodName">
            <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9_]*$"/>
            <message key="name.invalidPattern" value="Method name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="EmptyCatchBlock">
            <property name="exceptionVariableName" value="expected"/>
        </module>

        <!-- No need for Javadoc checks -->
        <module name="JavadocStyle">
            <property name="checkHtml" value="false"/>
            <property name="checkFirstSentence" value="false"/>
            <property name="severity" value="ignore"/>
        </module>
        <module name="JavadocType">
            <property name="severity" value="ignore"/>
        </module>
        <module name="JavadocMethod">
            <property name="severity" value="ignore"/>
        </module>
        <module name="JavadocVariable">
            <property name="severity" value="ignore"/>
        </module>
        <module name="SingleLineJavadoc">
            <property name="ignoreInlineTags" value="true"/>
            <property name="severity" value="ignore"/>
        </module>

        <!-- make comments available to filters -->
        <module name="FileContentsHolder"/>
    </module>
</module>
```

**Comment by jshiell** [comment]
Hmm. It seems fine here - your rules file really doesn't agree with my code (I use 4 spaces) but I can't find any false positives, and file navigation seems fine. This is with the dev branch, mind, which has some property fixes (and should fix @jrumbinas's problem).

Can you please tell me which version of IDEA you're using and on which platform? Can you also please tell me which version of Checkstyle you having selected?

For the record, my testing is on IDEA 2013.3.4, macOS Sierra, with CS 7.5.

**Comment by cjr8020** [comment]
Maven project was using CS 6.9, updated to CS 7.5, pulls config from URL served off our CI box
both checkstyle:check and checkstyle:checkstyle are clean with maven-checkstyle-plugin either 6.9 or 7.5

At the moment, I reverted back to Checkstyle-IDEA Version: 4.35.0
Scan result: "Checkstyle found no problems 
In IDEA, other settings > Checkstyle
Chechstyle version: 7.5
configuration file is pulled from the same URL

As soon as I upgrade my Checkstyle-IDEA plugin to 5.1.2, all my Java sources are in violation.  The list of files under "Checkstyle found 76 item(s) in 14 file(s)" shows no reason, and none of the files are clickable to navigate to source code.  Not the normal behavior, I don't think?

IntelliJ IDEA 2016.3.4
Build #IC-163.12024.16, built on January 31, 2017
JRE: 1.8.0_112-release-408-b6 x86
JVM: OpenJDK Server VM by JetBrains s.r.o

Windows 7 (don't ask)

**Comment by jshiell** [comment]
Thanks very much.

I've certainly found a problem - hopefully the problem, but we'll see. It looks like the result tree wasn't filtering ignored problems properly, so you'd get a message saying 'found x item(s) in x file(s)', but no nodes in the tree below them (as they were at a severity of ignored). This is fixed in 5.1.4, so that it'll correctly say no problems were found.

Can you please confirm if this makes any change for you, or if there's another problem I haven't spotted yet? Thank you.

**Comment by cjr8020** [comment]
5.1.4 seems to have addressed the issue of false positives.  Glad to help.
Thank you for maintaining this plugin!

**Comment by jrumbinas** [comment]
I do agree the message is gone, however editor is still marking ignored problems

**Comment by jshiell** [comment]
Thanks for the update - I have an idea what might be causing that - I'll try nail it later this week.

**Comment by jrumbinas** [comment]
Checkstyle also shows warning that files contain issues, although checkstyle panel indicates there're no problems.

**Comment by jshiell** [comment]
Hopefully *really* fixed in 5.2.0. But then I said that last time 😄
